test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+};
+
+// This stat is added later
+var fakeStat7 = {
+  stat_id: 7,
+  student_id: 2,
+  event_id: 2,
+  height: 320,
+  weight: 54,
+  pacer: 382,
+};
+
+// Update Checking
+var fakeStat8 = {
+  stat_id: 1,
+  student_id: 1,
+  event_id: 1,
+  height: 6,
+  weight: 6,
+  pacer: 6
+};
+
+describe('stats', function() {
+  describe('getStats(req)', function() {
+    xit('should get all the stats in the database', function(done) {
+      // GET all doesn't need anything from the request, so pass in empty
+      var promise = stats.getStats({});
+
+      // When the promised data is returned, check it against the expected data
+      promise.then(function(data) {
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6], data);
+        done();
+      });
+    });
+
+    // Get all stats at one site
+    xit('should get all stats for one site', function(done) {
+      var req = {
+        params: {
+          site_id: 1
+        }
+      };
+
+      var promise = stats.getStats(req);
+
+      promise.then(function(data) {
+        assert.deepEqual(data, [fakeStat, fakeStat4]);
+        done();
+      });
+    });
+
+    xit('should give an error if the site_id is negative',
+    function(done) {
+      var req = {
+        params: {
+          site_id: -4
+        }
+      };
+
+      var promise = stats.getStats(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given site_id is of invalid format (e.g. not an integer or' +
+        ' negative)');
+
+        // TODO Uncomment these if you're making comlex errors, Jonah.
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'site_id');
+        // assert.equal(err.propertyValue, req.params.site_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    // test for error if site_id not an integer
+    xit('should give an error if the stat_id is not an integer',
+    function(done) {
+      var req = {
+        params: {
+          site_id: 'hi'
+        }
+      };
+
+      var promise = stats.getStats(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given site_id is of invalid format (e.g. not an integer or' +
+        ' negative)');
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'site_id');
+        // assert.equal(err.propertyValue, req.params.site_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    // test for site_id not in database
+    xit('should give an error if the site_id is not in database',
+    function(done) {
+      var req = {
+        params: {
+          site_id: 44555555
+        }
+      };
+
+      var promise = stats.getStats(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given site_id not found in database.');
+        assert.equal(err.status, 400);
```

Comment:
This should not be an error at all. I think that we should just return the query and if it is an empty set that is for the front-end to deal with, not us.